### PR TITLE
Exclude wordpress-seo from vendor directory when developing.

### PR DESCRIPTION
**Pull request by hansjovis**
## Context

* We want to have a working development environment for WooCommerce SEO.

## Summary

This PR can be summarized in the following changelog entry:

* Fixes a bug in the WooCommerce SEO dev version where activating the WooCommerce SEO plugin would throw a fatal error. 

## Relevant technical choices:

* Excludes WordPress SEO from the `vendor` directory from the autoloader, since we only need that dependency for the tests.
* However, to make them work for the tests again, we need to:
	* Generate the autoload file for WordPress SEO in the vendor directory.
	* Load that autoload file before running the tests.  
* This PR also excludes the `vendor/yoast/wordpress-seo` directory from polluting the language POT file.

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

* Checkout this branch.
* Run `composer install`.
	* This should also run `composer install` in the `vendor/yoast/wordpress-seo` directory.
* Run `composer test` to make sure that the tests still work as expected.
* On your WordPress site: activate the WooCommerce SEO plugin. It should not throw a fatal error. 

### Test instructions for QA when the code is in the RC

* [ ] QA should use the same steps as above.

QA can test this PR by following these steps:

* No specific testing necessary, this bug only impacts development environments. 

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes

* [ ] This PR changes the UI in the plugin. I have added the 'UI change' label to this PR.

## Documentation

* [ ] I have written documentation for this change.

## Quality assurance

* [ ] I have tested this code to the best of my abilities
* [ ] I have added unittests to verify the code works as intended

Fixes #

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -39,7 +39,7 @@ jobs:
         - eval $(ssh-agent -s)
         - ssh-add ./deploy_keys/wpseo_woo_deploy
         - ssh-keyscan -t rsa github.com >> ~/.ssh/known_hosts
-        - composer config --global github-oauth.github.com $GITHUB_OAUTH_TOKEN
+        - composer config --global github-oauth.github.com ${GITHUB_OAUTH_TOKEN}
```
Comment (hansjovis) [inline .travis.yml:42]: For the person doing the CR: The curly braces are not strictly needed, but it is how we set the token in the other plugins, so I decided to leave this change in.

File: .travis.yml [hunk 2]
```diff
@@ -76,7 +76,26 @@ cache:
...
+      phpenv local 7.1
+      travis_retry composer install --no-interaction --ignore-platform-reqs
+      if [[ ${TRAVIS_PHP_VERSION:0:1} == "5" ]]; then
+        cd vendor/yoast/wordpress-seo
+        travis_retry composer remove humbug/php-scoper --dev --ignore-platform-reqs
+        travis_retry composer remove atanamo/php-codeshift --dev --ignore-platform-reqs
+        # Dependency injection compilation only works on PHP 7+.
+        # It makes use of reflection-based PHP code that only works on PHP 7+.
+        rm -rf vendor
+        travis_retry composer du
```
Comment (hansjovis) [inline .travis.yml:94]: For the person doing the CR: Removing the `vendor` directory is necessary to remove dependency conflicts. However, this also removes the autoload files for WordPress SEO, which we need to run the unit tests for WooCommerce SEO.

File: .travis.yml [hunk 3]
```diff
@@ -89,7 +89,9 @@ install:
     if [[ ${TRAVIS_PHP_VERSION:0:1} == "8" || $TRAVIS_PHP_VERSION == "nightly" ]]; then
       travis_retry composer install --no-interaction --ignore-platform-reqs
     elif [[ "$PHPCS" == "1" || "$PHPUNIT" == "1" || "$PHPLINT" == "1" ]]; then
+      phpenv local 7.1
       travis_retry composer install --no-interaction
+      phpenv local --unset
```
Comment (hansjovis) [inline .travis.yml:94]: For the person doing the CR: We need PHP 7 to be able to generate the WordPress SEO DI-container, see the `post-install-cmd` in `composer.json` so we can use it in the unit tests.

File: .travis.yml [hunk 4]
```diff
@@ -171,7 +173,7 @@ script:
   - |
     if [[ "$PHPUNIT" == "1" ]] && [[ ${TRAVIS_PHP_VERSION:0:1} == "8" || $TRAVIS_PHP_VERSION == "nightly" ]]; then
       travis_fold start "PHP.tests" && travis_time_start
-      travis_retry composer remove --dev phpunit/phpunit --no-update &&
+      travis_retry composer require --dev phpunit/phpunit:"^9.0" --update-with-dependencies --ignore-platform-reqs --no-interaction
```
Comment (hansjovis) [inline .travis.yml:176]: For PHP 8 we need to update the PHPUnit version (we do the same in the other plugins for the PHP 8 Travis build).

File: config/grunt/task-config/makepot.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+				poedit: true,
+				"report-msgid-bugs-to": "<%= pkg.pot.reportmsgidbugsto %>",
+				"language-team": "<%= pkg.pot.languageteam %>",
+				"last-translator": "<%= pkg.pot.lasttranslator %>",
+			},
+			type: "wp-plugin",
+			exclude: [
+				"<%= files.artifact %>",
+				".wordpress-svn",
+				"vendor/yoast/wordpress-seo",
```
Comment (hansjovis) [inline config/grunt/task-config/makepot.js:17]: We need to specifically exclude the `wordpress-seo` from contaminating the translation files.

**Comment by Xyfi** [comment]
Acceptance 👍

**Pull request merged**